indico/web/client/js/react/components/principals/PrincipalListField.jsx

```diff
@@ -156,6 +163,20 @@ const PrincipalListField = props => {
               onChange={(e, data) => handleAddItems([{identifier: data.value}])}
             />
           )}
+          {registrationForms.length !== 0 && (
+            <Dropdown
```

Comment:
Now that we have 3 of these dropdowns, what about wrapping it in a component?

```jsx
<AddPrincipalDropdown
  text={Translate.string('Registration Form')}
  options={registrationFormOptions}
  onAdd={value => handleAddItems([{identifier: value}])}
/>
```

Reply:
Sounds good, didn't even notice this. Much better! I tried to build a map but the text doesn't help. Didn't move the `ifs` because it feels weird to let the child control rendering.